pandas/tools/tests/test_util.py

```diff
@@ -291,6 +291,76 @@ def test_non_hashable(self):
         with self.assertRaisesRegexp(TypeError, "Invalid object type"):
             pd.to_numeric(s)
 
+    def test_downcast(self):
+        mixed_data = ['1', 2, 3]
+        int_data = [1, 2, 3]
+        date_data = np.array(['1970-01-02', '1970-01-03',
+                              '1970-01-04'], dtype='datetime64[D]')
+
+        smaller_dtypes = [np.int8, np.uint8, np.int16, np.uint16,
+                          np.int32, np.uint32, np.float32]
+
+        larger_dtypes = [np.int64, np.uint64, np.float64]
+
+        msg = "'downcast' must be a numerical dtype"
```

Comment:
I disagree.  That's a misleading error message if they pass in `downcast=datetime64[D]`.  My current error message is broad enough to encompass both invalid dtypes and non-numeric ones.